sources/packages/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -203,4 +209,76 @@ export class ApplicationControllerService {
       );
     }
   }
+
+  /**
+   * Process application income verification details.
+   * @param CRAIncomeVerification CRA income verification data.
+   * @return processed object.
+   */
+  processApplicationIncomeVerificationDetails(
+    CRAIncomeVerification: CRAIncomeVerification[],
+  ): ApplicationIncomeVerification {
+    const incomeVerificationDetails = {} as ApplicationIncomeVerification;
+    const studentIncomeVerifications = CRAIncomeVerification.filter(
+      (incomeVerification) => !incomeVerification.supportingUser,
+    );
+    const supportingUserIncomeVerification = CRAIncomeVerification.filter(
+      (incomeVerification) => incomeVerification.supportingUser,
+    );
+    const [studentIncomeVerification] = studentIncomeVerifications;
+    // Student income verification details.
+    incomeVerificationDetails.studentIncomeVerificationStatus = {
+      waiting: !studentIncomeVerification.dateReceived,
+      success: !!studentIncomeVerification.dateReceived,
+    };
+
+    supportingUserIncomeVerification.forEach((incomeVerification, index) => {
+      // Supporting user income verification details.
+      if (
+        incomeVerification.supportingUser.supportingUserType ===
+        SupportingUserType.Parent
+      ) {
+        // If supporting user type is parent, then there can be 1 or 2 parent.
+        incomeVerificationDetails[
```

Comment:
Because the object is not dynamic, I would try to ensure the type safety and do something like below:
```ts
const supportingUserParents = CRAIncomeVerification.filter(
      (incomeVerification) =>
        incomeVerification.supportingUser?.supportingUserType ===
        SupportingUserType.Parent,
    );
    const [parent1, parent2] = supportingUserParents;
    if (parent1) {
      incomeVerificationDetails.parent1IncomeVerificationStatus = {
        waiting: !parent1.dateReceived,
        success: !!parent1.dateReceived,
      } as SuccessWaitingStatus;
    }

    if (parent2) {
      incomeVerificationDetails.parent2IncomeVerificationStatus = {
        waiting: !parent1.dateReceived,
        success: !!parent1.dateReceived,
      } as SuccessWaitingStatus;
    }
```

Reply:
@ann-aot please note that there is an error in the above sample where the paren1 was consumed in the parent2 logic.